src/braket/pennylane_plugin/braket_device.py

```diff
@@ -270,18 +266,14 @@ def __init__(
             raise ValueError(f"Device {device.name} does not support quantum circuits")
 
         device_type = device.type
-        if shots is not None:
-            if shots == 0 and device_type == AwsDeviceType.QPU:
-                raise ValueError("QPU devices do not support 0 shots")
-            num_shots = shots
-        elif device_type == AwsDeviceType.SIMULATOR:
-            num_shots = AwsDevice.DEFAULT_SHOTS_SIMULATOR
-        elif device_type == AwsDeviceType.QPU:
-            num_shots = AwsDevice.DEFAULT_SHOTS_QPU
-        else:
+
+        if shots is None and device_type == AwsDeviceType.QPU:
+            raise ValueError("QPU devices require the number of shots to be specified")
+
+        if not (device_type == AwsDeviceType.SIMULATOR or device_type == AwsDeviceType.QPU):
```

Comment:
I wasn't sure if we still needed this check?

Reply:
As with the comment on line 227, device-specific defaults should be used